checkstyle/suppressions.xml

```diff
@@ -51,13 +51,13 @@
               files="(Utils|Topic|KafkaLZ4BlockOutputStream|AclData).java"/>
 
     <suppress checks="CyclomaticComplexity"
-              files="(ConsumerCoordinator|Fetcher|Sender|KafkaProducer|BufferPool|ConfigDef|RecordAccumulator|KerberosLogin|AbstractRequest|AbstractResponse|Selector|SslFactory|SslTransportLayer|SaslClientAuthenticator|SaslClientCallbackHandler).java"/>
+              files="(ConsumerCoordinator|Fetcher|Sender|KafkaProducer|BufferPool|ConfigDef|RecordAccumulator|KerberosLogin|AbstractRequest|AbstractResponse|Selector|SslFactory|SslTransportLayer|SaslClientAuthenticator|SaslClientCallbackHandler|TopologyOptimizerImpl|StreamsTopologyGraphImpl|JoinGraphNode).java"/>
```

Comment:
Should we reconsider adding these supressions? These restrictions help prevent us from writing unmaintainable code. Maybe it's a sign that we should break up some of these classes or add utilities to simplify the top-level logic.